### PR TITLE
Us23 01 secure form

### PR DESCRIPTION
**Pull request by Wowlfy**

### CODE REVIEW

**Review by Loic-Code — DISMISSED** [review]

File: src/Controller/ContactController.php [hunk 1]
```diff
@@ -6,6 +6,41 @@ class ContactController extends AbstractController
...
     public function contact(): string
     {
-        return $this->twig->render('User/contact.html.twig');
+        $message = '';
+        $data = [];
+        $errors = [];
+
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $data = array_map('trim', $_POST);
+
```
Comment (LucyMerlier) [inline src/Controller/ContactController.php:16]: there must be length validations for each of your fields

File: src/View/User/Includes/_contact_grey_block.html.twig [hunk 2]
```diff
@@ -2,25 +2,41 @@
...
+            {% if errors %}
+                <ul class="alert alert-danger error">
+                    {% for error in errors %}
+                        <li>{{ error }}</li>
+                    {% endfor %}
+                </ul>
+            {% elseif errors is empty and data is null %}
+                <ul class="alert alert-success validated">
+                    <li>{{ success }}</li>
+                </ul>
```
Comment (LucyMerlier) [inline src/View/User/Includes/_contact_grey_block.html.twig]: why a list just for one item?

File: src/View/User/Includes/_contact_grey_block.html.twig [hunk 3]
```diff
@@ -2,25 +2,41 @@
...
+                    <label for="firstname" class="form-label">
+                        <span class="text-danger">*</span>Prénom</label>
+                    <input type="text" class="form-control" id="firstname" name="firstname" placeholder="Delphine" aria-required="true">
                 </div>
                 <div class="mb-3">
-                    <label for="phone" class="form-label">Telephone</label>
-                    <input type="text" class="form-control" id="phone" name="phone">
+                    <label for="phone" class="form-label">
+                        <span class="text-danger">*</span>Téléphone</label>
+                    <input type="text" class="form-control" id="phone" name="phone" placeholder="06.00.00.00.00." aria-required="true">
```
Comment (LucyMerlier) [inline src/View/User/Includes/_contact_grey_block.html.twig]: there is a type="tel" that is basically a type="text" but has better UX for phone users

File: src/View/User/Includes/_contact_grey_block.html.twig [hunk 4]
```diff
@@ -2,25 +2,41 @@
...
-                    <label for="email"class="form-label">Email</label>
-                    <input type="email" class="form-control" id="email" name="email">
+                    <label for="email" class="form-label">
+                        <span class="text-danger">*</span>Email</label>
+                    <input type="email" class="form-control" id="email" name="email" placeholder="jean.tie@poney.com" aria-required="true">
                 </div>
                 <div class="mb-3">
-                    <label for="message" class="form-label">Message</label>
-                    <textarea class="form-control" id="message" name="textarea" rows="3"></textarea>
+                    <label for="email" class="form-label">
```
Comment (LucyMerlier) [inline src/View/User/Includes/_contact_grey_block.html.twig]: wrong "for" attribute => should be "message"

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments + why are there changes in composer.lock?

**Review by bouboumael — APPROVED** [review]

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
composer lock shouldnt have had any modifications

**Pull request closed without merging**